mangaki/mangaki/utils/mal.py

```diff
@@ -220,18 +229,72 @@ def search_works(self, work_type: MALWorks, query: str) -> List[MALEntry]:
                    getattr(settings, 'MAL_PASS', None))
 
 
-def insert_into_mangaki_database_from_mal(mal_entries: List[MALEntry]):
+def look_for_works(work_list: QuerySet,
+                   ext_poster: str,
+                   titles: List[str]) -> List[Work]:
+    constraints_work = None
+    constraints_title = None
+    for synonym in titles:
```

Comment:
Much agreed, most of this complexity stayed here due to former attempts to improve performance.